src/meltano/cli/schedule.py

```diff
@@ -27,57 +32,154 @@ def schedule(project, ctx):
 
 
-@schedule.command(short_help="[default] Add a new schedule.")
-@click.argument("name")
-@click.argument("extractor")
-@click.argument("loader")
-@click.argument("interval")
-@click.option("--transform", type=click.Choice(["skip", "only", "run"]), default="skip")
-@click.option("--start-date", type=click.DateTime(), default=None)
-@click.pass_context
-def add(ctx, name, extractor, loader, transform, interval, start_date):
-    """
-    Add a new schedule.
-
-    \b
-    NAME:\tThe schedule name, must be unique
-    EXTRACTOR:\tWhich extractor should be used
-    LOADER:\tWhich loader should be used
-    INTERVAL:\tCron-like syntax to specify the schedule interval (@daily, @hourly, etc…)
-    """
+def _add_elt(
+    ctx,
+    name: str,
+    extractor: str,
+    loader: str,
+    transform: str,
+    interval: str,
+    start_date: str | None,
+):
+    """Add a new legacy elt schedule."""
     project = ctx.obj["project"]
-    schedule_service = ctx.obj["schedule_service"]
+    schedule_service: ScheduleService = ctx.obj["schedule_service"]
 
-    _, Session = project_engine(project)  # noqa: N806
-    session = Session()
+    _, session_maker = project_engine(project)
+    session = session_maker()
     try:
-        added_schedule = schedule_service.add(
+        added_schedule = schedule_service.add_elt(
             session, name, extractor, loader, transform, interval, start_date
         )
+        tracker = GoogleAnalyticsTracker(schedule_service.project)
+        tracker.track_meltano_schedule("add", added_schedule)
+        click.echo(
+            f"Scheduled elt '{added_schedule.name}' at {added_schedule.interval}"
+        )
+    except ScheduleAlreadyExistsError:
+        click.secho(f"Schedule '{name}' already exists.", fg="yellow")
+    finally:
+        session.close()
+
+
+def _add_job(ctx, name: str, job: str, interval: str):
+    """Add a new scheduled job."""
+    project = ctx.obj["project"]
+    schedule_service: ScheduleService = ctx.obj["schedule_service"]
 
+    _, session_maker = project_engine(project)
+    session = session_maker()
+    try:
+        added_schedule = schedule_service.add(name, job, interval)
         tracker = GoogleAnalyticsTracker(schedule_service.project)
         tracker.track_meltano_schedule("add", added_schedule)
-        click.echo(f"Scheduled '{added_schedule.name}' at {added_schedule.interval}")
-    except ScheduleAlreadyExistsError as serr:
-        click.secho(
-            f"Schedule '{serr.added_schedule.name}' already exists.", fg="yellow"
+        click.echo(
+            f"Scheduled job '{added_schedule.name}' at {added_schedule.interval}"
         )
+    except ScheduleAlreadyExistsError:
+        click.secho(f"Schedule '{name}' already exists.", fg="yellow")
     finally:
         session.close()
 
 
+@schedule.command(short_help="[default] Add a new schedule.")
+@click.argument("name")
+@click.option("--interval", "-i", required=True, help="Interval of the schedule.")
```

Comment:
I added both long and short flags, happy to remove the short flags - but fwiw - its definitely nice not having to type them out.

Reply:
@pandemicsyn - I do like the short flags. However, I want to be careful about the namespace of known flags and try to avoid collisions down the road as we add more.

- `-i` could be `--interval` or `--interactive`
- `-e` could be `--extractor` or `--environment`
- `-l` could be `--loader` or `--log-level`

What do you think? 

cc @tayloramurphy

Reply:
@pandemicsyn how about we just leave it off for now so as to unblock this. Can you then open a discussion (or even ADR) on how we want to use short flags?